coati/builder.py

```diff
@@ -107,8 +120,36 @@ def mergeresources(self, resources):
             merge.resources(slide, objs)
 
     def build(self):
-        self.fillexcel()
-        resources = self.resourcedict() or dict()
-        self.loadtemplate()
-        self.mergeresources(resources)
-        self.closexcel()
+        tslides = self._pptx.Slides()
+
+        for number, params in self.slidesdata:
+            slide = tslides.Item(number)
+            self.process_slide(slide, params)
+
+    def process_slide(self, slide, params):
+        if type(params) is not list:
+            params = [params]
+
+        def process(slide, shapename, stuple):
+            stype = stuple[0]
+            self._processfunctions[stype](slide, shapename, stuple )
+
+        def checktype(stuple):
+            stype = stuple[0]
+            return stype in self._processfunctions
+
+        return [process(slide, k, d[k]) for d in params for k in d if checktype(d[k])]
+
+    def _processlabel(self, slide, shapename, stuple):
```

Comment:
Is not the responsibility of the SlideBuilder to know how the resources are created